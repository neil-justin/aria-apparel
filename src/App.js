import './styles/header.css';
import './styles/index.css';
import './styles/home.css';
import './styles/shop.css';
import { Route, Routes, NavLink } from 'react-router-dom';
import Home from './components/Home';
import Shop from './components/Shop';

function App() {
  return (
    <div className="App">
      <header id='app-header'>
        <span id="logo">Aria Apparel</span>

        <nav id='header-nav'>
          <ul>
            <li><NavLink to='/'>Home</NavLink></li>
            <li><NavLink to='/shop'>Shop</NavLink></li>
          </ul>
        </nav>
      </header>

      <Routes>
        <Route path='/' element={<Home />} />
        <Route path='/shop' element={<Shop />} />
      </Routes>
    </div>
  );
}

export default App;