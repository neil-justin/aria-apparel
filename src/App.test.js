import React from "react";
import { act, getByText, render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom"; // import the Router component
import App from "./App";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";

describe('App component', () => {
    it('should render the home page by default', () => {
        const { container } = render(
            <Router>
                <App />
            </Router>
        );

        expect(container).toMatchSnapshot();
    });

    it('should render the shop page after the Shop NavLink is clicked', async () => {
        render(
            <Router>
                <App />
            </Router>
        );
        await act(async () => {
            userEvent.click(screen.getByRole('link', { name: 'Shop' }));
        });
        await waitFor(() => screen.getByRole('heading', { name: 'Products' }));
        expect(screen.getByRole('heading', { name: 'Products' })).toBeInTheDocument();
    });
})