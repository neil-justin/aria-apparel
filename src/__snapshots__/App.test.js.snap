// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render the home page by default 1`] = `
<div>
  <div
    class="App"
  >
    <header
      id="app-header"
    >
      <span
        id="logo"
      >
        Aria Apparel
      </span>
      <nav
        id="header-nav"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class=""
              href="/shop"
            >
              Shop
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      id="home-main-content"
    >
      <div
        id="hero"
      >
        <h1>
          Welcome to Aria Apparel!
        </h1>
        <p
          id="hero-para"
        >
          Shop with ease and convenience at Aria Apparel. Find unbeatable prices on high-quality products, all in one convenient location
        </p>
        <button
          id="hero-button"
        >
          Shop now
        </button>
      </div>
    </main>
  </div>
</div>
`;
